event/event.go

```diff
@@ -0,0 +1,181 @@
+// event is a package that models events that occur during the execution of a program
+package event
+
+// Event is an interface that custom events should implement.
+// It is empty for now because there are no general methods
+// required of events yet.
+type Event interface {
+}
+
+// KeyCharacter is an event that occurs when a character is actually typed on
+// the keyboard. This may be provided by an input method.
+type KeyCharacter struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+	// Character typed
+	Character rune
+}
+
+// KeyDown is an event that occurs when a key is pressed on the keyboard.
+type KeyDown struct {
+	// Code of the key pressed or released
+	Code int
+	// Key board modifiers
+	Modifiers int
+}
+
+// KeyUp is an event that occurs when a key is released on the keyboard.
+// The data is the same as for a KeyDown event
+type KeyUp KeyDown
+
+// GamepadAxis is for event where an axis on a game pad changes.
+type GamepadAxis struct {
+	// Which gamepad caused the event
+	ID int
+	// Axis of the game pad that changed, if any
+	Axis int
+	// Position of the axis after the change
+	Position float32
+}
+
+//GamepadButtonDown is a game pad button press event.
+type GamepadButtonDown struct {
+	// Which gamepad caused the event
+	ID int
+	// Button that was pressed, if any.
+	Button int
+	// Position of the button after the change.
+	Position float32
+}
+
+//GamepadButtonDown is a game pad button release event.
+//The data is identical to a GamePadButtonDown event.
+type GamepadButtonUp GamepadButtonDown
+
+//GamepadAttach happens when a new game pad is attached.
+type GamepadAttach struct {
+	// Which gamepad caused the event
+	ID int
+	// Amount of axes the game pad has.
+	Axes int
+	// Amount of buttons it has.
+	Buttons int
+}
+
+//GamepadDetach happens when a game pad is detached.
+type GamepadDetach struct {
+	// Which gamepad caused the event
+	ID int
+}
+
+//MouseAxes is a mouse axis event.
+type MouseAxis struct {
+	// X position of the event
+	X float32
+	// Y position of the event
+	Y float32
+	// Wheel is the position of the mouse wheel
+	Wheel float32
+	// DeltaX is the change in X since last event
+	DeltaX float32
+	// DeltaY is the change in Y since last event
+	DeltaY float32
+	// DeltaWheelis the change in the wheel position since last event
+	DeltaWheel float32
+}
+
+//MouseButtonDown is a mouse button press event.
+type MouseButtonDown struct {
+	// X position of the event
+	X float32
+	// Y position of the event
+	Y float32
+	// Wheel is the position of the mouse wheel
+	Wheel float32
+	// DeltaX is the change in X since last event
+	DeltaX float32
+	// DeltaY is the change in Y since last event
+	DeltaY float32
+	// DeltaWheelis the change in the wheel position since last event
+	DeltaWheel float32
+	// Button that was pressed.
+	Button int
```

Comment:
What kind of value is used here?

Reply:
It's the number of the mouse button, much like the number of the gamepad button. Nowadays, some gaming mice have up to 15 mouse buttons, so not just 3 buttons.  Would you like this too to become an enum of MouseButton, with MouseButtonLeft=1, MouseButtonRight=2, MouseButtonMiddle=3,   MouseButtonExtra1 = 4 , etc  or such?

Reply:
We should have enums. Please leave a TODO comment same as Keys :)

Reply:
OK, will do!